src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java

```diff
@@ -18,6 +20,16 @@ public StorageSharedKeyCredential getStorageSharedKeyCredential(
         return new StorageSharedKeyCredential(accountName, accountKey);
     }
 
+    // not used as needs test context so it can actually be build
+    @Bean({
+        "storage-client",
+        "bulkscan-storage-client",
+        "crime-storage-client"
+    })
+    public BlobServiceClient getStorageClient(StorageSharedKeyCredential credentials) {
+        return new BlobServiceClientBuilder().credential(credentials).buildClient();
+    }
```

Comment:
if those beans are not used, do they need to be defined? (and point to the same container)

Reply:
Need to be defined to load application context for certain integration tests. Otherwise:

> org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.azure.storage.blob.BlobServiceClient' available

This was achieved by just returning a `null` here as well

Reply:
right, it would probably be best to limit the beans to the minimum that the tests need